plugin/Controller/AdminViewController.php

```diff
@@ -427,19 +428,29 @@ public function generalAdminNoticeSEO() {
 		$urlOnofficeSetting = admin_url().'admin.php?page=onoffice-settings';
 		$nameOnofficeSetting = esc_html__('onOffice plugin settings','onoffice-for-wp-websites');
 		$pluginOnofficeSetting = sprintf("<a href='%s' target='_blank' rel='noopener'>%s</a>", $urlOnofficeSetting,$nameOnofficeSetting);
+		$pluginSEO = [];
+		if (count(array_intersect(["wpseo/wp-seo.php"], get_option("active_plugins"))) > 0) {
+			$pluginSEO['wp_seo'] = 'wpSEO';
+		}
+		if (count(array_intersect(["seo-by-rank-math/rank-math.php"], get_option("active_plugins"))) > 0) {
+			$pluginSEO['rank_math_seo'] = 'Rank Math SEO';
+		}
+		if (count(array_intersect(["wordpress-seo/wp-seo.php"], get_option("active_plugins"))) > 0) {
+			$pluginSEO['yoast_seo'] = 'Yoast SEO';
+		}
+		$listNamePluginSEO = implode(", ", $pluginSEO);
 		if (count(array_intersect(["wordpress-seo/wp-seo.php", "seo-by-rank-math/rank-math.php", "wpseo/wpseo.php"], get_option("active_plugins"))) > 0) {
 			if (get_option('onoffice-click-button-close-action') == 0
 				&& get_current_screen()->id !== 'onoffice_page_onoffice-settings'
 				&& get_option('onoffice-settings-title-and-description') == 0) {
-				echo "<div class='notice notice-warning active-plugin-seo is-dismissible'>
-						<p> " . esc_html__('The onOffice plugin has detected an active SEO plugin: Yoast SEO.', 'onoffice-for-wp-websites') . "
-							" . esc_html__('You currently have configured the onOffice plugin to fill out the title and description of the detail page, which can lead to conflicts with the SEO plugin.', 'onoffice-for-wp-websites') . "<br>
-							" . esc_html__('We recommend that you go to the', 'onoffice-for-wp-websites') . "
-							" . $pluginOnofficeSetting . "
-							" . esc_html__('and configure the onOffice plugin to not modify the title and description.', 'onoffice-for-wp-websites') . "
-							" . esc_html__('This allows you to manage the title and description with your active SEO plugin.', 'onoffice-for-wp-websites')
-						. "</p>
-					</div>";
+				$class = 'notice notice-warning active-plugin-seo is-dismissible';
+				$message = sprintf(esc_html__('The onOffice plugin has detected an active SEO plugin: %s. You currently have configured the onOffice plugin to fill out the title and description of the detail page, which can lead to conflicts with the SEO plugin.
+								We recommend that you go to the %s and configure the onOffice plugin to not modify the title and description. This allows you to manage the title and description with your active SEO plugin.', 'onoffice-for-wp-websites'), $listNamePluginSEO, $pluginOnofficeSetting);
+				$message = Parsedown::instance()
```

Comment:
Please keep in mind injection vulnerabilities and use [Parsedown's safe mode](https://github.com/erusev/parsedown#security).

Reply:
> A few notes and an addition.
> 
> The addition (it was not in the requirements) is to add some more SEO plugins. I've [searched for SEO plugins](https://wordpress.org/plugins/search/seo/) and would like to add those I found with over 100,000 installs.
> 
> The updated list is:
> 
> * `wordpress-seo/wp-seo.php` -> "Yoast SEO"
> * `seo-by-rank-math/rank-math.php` -> "Rank Math SEO"
> * `wpseo/wpseo.php` -> "wpSEO"
> * `all-in-one-seo-pack/all_in_one_seo_pack.php` -> "All in One SEO"
> * `autodescription/autodescription.php` -> "SEO Framework"
> * `wp-seopress/seopress.php` -> "SEOPress"
> * `squirrly-seo/squirrly.php` -> "Squirrly SEO"

@jmaas-onoffice I tested with The updated Seo plugin list with "custom field(4.2)" feature
A. Only this plugin allows support:
1.wordpress-seo/wp-seo.php -> "Yoast SEO" (ex:%%_cf_onoffice_title%%)

2.seo-by-rank-math/rank-math.php -> "Rank Math SEO"(ex:%customfield(onoffice_description)%)

3.wp-seopress/seopress.php -> "SEOPress"(ex:%%_cf_onoffice_title%%)

4.all-in-one-seo-pack/all_in_one_seo_pack.php -> "All in One SEO"(ex:Custom Field-onoffice_description )

B.This plugin allows do not support
5.wpseo/wpseo.php -> "wpSEO" 
(The sale of the wpSEO plugin was discontinued due to a change of ownership.
wpSEO will soon be free
In the future, wpSEP will be available free of charge with the same range of functions.)

6.autodescription/autodescription.php -> "SEO Framework"
 https://theseoframework.com/changelog/4-0-0-multiplex/
(When using custom meta title or meta description filters, they may not be processed for the meta input fields.
Fixing this requires restructuring the custom title and description methods with flags.)

7.squirrly-seo/squirrly.php -> "Squirrly SEO"(https://howto.squirrly.co/kb/seo-automation/#title_automation)

Now, will we remove this plugin do not support custom field?

Reply:
@jmaas-onoffice Please give me your opinion on this matter, thanks

Reply:
So for wpSEO, SEO Framework, and Squirrly SEO we are not sure whether they can use custom fields?

It's actually ok if they can't. The warning is also about preventing double `<title>` and `<meta name="description">`. So it's worth keeping the warning for those plugins so that people can deactivate the plugin's `<title>` and `<meta>` so that there is only one. If their SEO plugin does not have the functions that they need, they will have to change to another SEO plugin.

wpSEO not being sold anymore is also ok. Some people might still have it and it is just one line in our code that can't break anything, right?

But thank you for your tests and research, especially the links to the documentation! It was a good idea to check it, I didn't know that some of them don't appear to support custom fields!